### ISSUE TITLE
Drilldown links are broken (404)

### ISSUE THREAD

**Issue by fculpo** [kubernetes-monitoring/kubernetes-mixin#659]
Hi,

I just installed a fresh kube-prometheus instance.
Grafana is served on an ingress like https://grafana.xxx.net (no subpath)

All drilldown links are broken, I always get a 404 error.

![Capture d’écran 2021-08-11 à 11 35 11](https://user-images.githubusercontent.com/325117/129006310-9930d9c6-a6ef-46b2-a982-fa83a7e9e768.png)
![Capture d’écran 2021-08-11 à 11 35 28](https://user-images.githubusercontent.com/325117/129006326-a0a69748-2f19-4d58-88e8-461f3c650629.png)

**Comment by fculpo** [comment]
Parent dashboard is for instance: https://grafana.xxx.net/d/efa86fd1d0c121a26444b636a3f509a8/kubernetes-compute-resources-cluster?orgId=1&refresh=10s&var-datasource=Cortex&var-cluster=monitoring

Link points to https://grafana.xxx.net/d/85a562078cdf77779eaa1add43ccec1e/k8s-resources-namespace?var-datasource=Cortex&var-cluster=monitoring&var-namespace=ingress-nginx

**Comment by fculpo** [comment]
If I try to access manually the child link, this seems to redirect to https://grafana.xxx.net/d/85a562078cdf77779eaa1add43ccec1e/kubernetes-compute-resources-namespace-pods?var-datasource=Cortex&var-cluster=monitoring&var-namespace=kube-system&orgId=1&refresh=10s

**Comment by chdxD1** [comment]
Noticed the same in our deployment with Grana 8. The issue seems to be the `./` in front of the links in the dasboards (coming from the linkPrefix). Not sure what happens if that bit will be removed, especially with Grafana running on subpaths (see https://github.com/kubernetes-monitoring/kubernetes-mixin/issues/192)

After removing the ./ from our dashboards it worked again in Grafana 8. There is also an issue with the slugs as @fculpo has mentioned which could be fixed as well.

This is probably related to the move of Grafana 8 from Angular Router to react Router?

**Comment by sdelrio** [comment]
If you click reload when getting the 404 (the url in browser seems correct) then the drill down target works. So agree it looks like a router issue

**Comment by zerda** [comment]
Problem still exists in Grafana 8.1.2, there is a related issue but closed. https://github.com/grafana/grafana/issues/36532

When right click the link open in new tab, it's working fine. when left click the link, it's rendered in the SPA (no refresh page at all) with 404 Error.

BTW from [docs](https://grafana.com/docs/grafana/latest/http_api/dashboard/) says, grafana account the dashboard uid only, name is ignored whether /k8s-resources-namespace or /kubernetes-compute-resources-namespace-pods.

**Comment by philimonoff** [comment]
I've found, that there are characters "./" in some linkUrls, and there are not in many others. But I don't know the reason of this difference.

**Comment by philimonoff** [comment]
@fculpo I had the same problem and I've found the reason. There is a prefix variable [here](https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/master/dashboards/resources/multi-cluster.libsonnet#L12). And it fills from "linkPrefix" if [config](https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/master/config.libsonnet#L72). It was enough for me to overwrite default value `linkPrefix: '.'` with `linkPrefix: ''` and everything is okay now.

**Comment by mariaaloysiusa** [comment]
@philimonoff - could you please let me know the exact path on how to overwrite the vaule of linkPrefix

**Comment by philimonoff** [comment]
@mariaaloysiusa it depends on how exactly you run mixin. If you just cloned the repo of mixin, you should change it in [config.libsonnet](https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/master/config.libsonnet#L72) and regenerate your dashboards. If you import mixin as a lib, you should add a `config` object.
```json
local kubernetes = import 'kubernetes-mixin/mixin.libsonnet';

kubernetes {
  _config+:: {
    grafanaK8s+:: {
      grafanaTimezone: 'browser',
      linkPrefix: ''
    },
  },
}
```

**Comment by fculpo** [comment]
works perfectly

We may close this ?

**Comment by paulfantom** [comment]
Fixed in #701
